api/upload.py

```diff
@@ -204,14 +255,42 @@ def engine(self):
             self.abort(400, 'container id is required')
         else:
             cid = bson.ObjectId(cid)
+
         context = {
             'job_id': self.get_param('job'),
             'job_ticket_id': self.get_param('job_ticket'),
         }
-        if level == 'analysis':
-            return process_upload(self.request, Strategy.analysis_job, origin=self.origin, container_type=level, id_=cid, context=context)
+
+        # Request for upload ticket
+        if self.get_param('ticket') == '':
+            return self._create_ticket()
+
+        # Check ticket id and skip permissions check if it clears
+        ticket_id = self.get_param('ticket')
+        if ticket_id:
+            ticket = self._check_ticket(ticket_id)
+            if not self.origin.get('id'):
+                # If we don't have an origin with this request, use the ticket's origin
```

Comment:
In what situation would you expect us to not have an origin? Above we check to make sure the request has admin rights, so we only allow authenticated requests for this endpoint.